aspnetcore/signalr/introduction-signalr-core.md

```diff
@@ -0,0 +1,60 @@
+ms.author: rachelap
+description: An overview of SignalR on ASP.NET Core
+manager: wpickett
+ms.date: 2/19/2018
+ms.topic: article
+ms.technology: dotnet-signalr
+ms.prod: aspnet-core
+---
+# Introduction to SignalR
+
+By [Rachel Appel](https://twitter.com/rachelappel)
+
+## What is SignalR?
+
+ASP.NET Core SignalR is a library that simplifies adding real-time web functionality to apps. Real-time web functionality is the ability to have server code push content to connected clients instantly as it becomes available, rather than having the server wait for a client to request new data.
+
+Good candidates for adding real-time updates include:
+
+* Apps that require high frequency updates from the server. Real-time gaming, social networks, maps and GPS apps, require high frequency updates.
+* Dashboards and monitoring apps. Some examples of these include voting and auction software.
+* Collaborative apps. Whiteboard apps and team meeting software are examples of collaborative apps.
+* Apps that require notifications. Social networks, email, chat, games, and many more types of apps use notifications.
+
+SignalR provides an API for creating server-to-client remote procedure calls (RPC). The RPCs call JavaScript functions on browsers or other client platforms from server-side .NET Core code. The SignalR API includes features such as connection management, groups,  messaging, and security.
+
+SignalR for ASP.NET Core:
+
+* Handles connection management automatically.
+* Enables broadcasting messages to all connected clients simultaneously. For example, a chat room.
+* Enables sending messages to specific clients or groups of clients.
+* Is open-sourced at [GitHub](https://github.com/aspnet/SignalR).
+* Scales nicely using [Azure Service Bus](https://azure.microsoft.com/en-us/services/service-bus/), SQL Server, or [Redis](http://redis.io).
+
+The connection between the client and server is persistent, unlike an HTTP connection. 
+
+## Transports and fallbacks
+
+ SignalR is an abstraction over some of the transports that are required to do real-time work between client and server. A SignalR connection starts as HTTP. SignalR uses the [WebSocket transport](https://tools.ietf.org/html/rfc7118) when available, and falls back to other transports when it's not available. WebSocket is the ideal transport for SignalR. This is because it makes the most efficient use of server memory, has the lowest latency, and has the most underlying features, such as full duplex communication between client and server. However, it also has the most stringent requirements: WebSocket requires the server to be using Windows Server 2012 or Windows 8, and .NET Framework 4.5, or later versions of these. If these requirements are not met, SignalR will attempt to use other transports to make its connections. The important thing to note is that SignalR automatically assigns a transport per connection for you automatically, though you can override this functionality.
```

Comment:
We should fix the requirements. We're now cross platform so we should be calling out what doesn't work. We need to do some more research here to see if these restrictions are all in place still. When running behind IIS, you need windows 8 and above and Server 2012, that isn't the case when using kestrel alone though. 

Also for the client side I believe we support we support websockets on win7 now (I'll confirm)

Reply:
> Also for the client side I believe we support we support websockets on win7 now (I'll confirm)

This is correct but **only** when using Kestrel as the edge server. If you're using IIS/IIS Express or HttpSys, WebSockets won't work on Windows 7.

Reply:
@anurse huh? What does that have to do with the client supporting websockets? (I'm talking about the C# client).